back/app/models/concerns/user_confirmation.rb

```diff
@@ -70,16 +70,12 @@ def increment_confirmation_retry_count!
   private
 
   def set_confirmation_required
-    return unless email_changed?
+    return unless new_record? && email_changed?
 
-    not_ordinary_user = highest_role != :user # admin or moderator can be created only by invite or by admin API from cl2-tenant-setup
-    confirmation_not_required =
-      sso? || invite_status.present? ||
-      active? || not_ordinary_user
+    return unless confirmation_required # to be able to create a confirmed user
 
+    confirmation_not_required = invite_status.present? || active?
     self.confirmation_required = !confirmation_not_required
-    self.email_confirmed_at = nil
-    self.email_confirmation_code_sent_at = nil
```

Comment:
Just wondering where these get set now

Reply:
Good question. It's not set now. `set_confirmation_required` is called only on create, so these two fields are always `nil` at this point anyway. And it's not possible to set them via controller because of `UserPolicy#permitted_attributes_for_create`.